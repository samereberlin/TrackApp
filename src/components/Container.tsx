import React from 'react';
import {StyleSheet, SafeAreaView, ViewStyle} from 'react-native';

import {colors} from '../utils/theme';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: colors.containerBackground,
  } as ViewStyle,
});

const Container: React.FC = ({children}) => (
  <SafeAreaView style={styles.container}>{children}</SafeAreaView>
);

export default Container;
