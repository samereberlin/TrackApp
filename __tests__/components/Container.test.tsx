import React from 'react';
import {create} from 'react-test-renderer';
import {Text} from 'react-native';
import {render} from '@testing-library/react-native';

import Container from '../../src/components/Container';

describe('Header', () => {
  it('matches the snapshot', () => {
    const screen = create(
      <Container>
        <Text>foo</Text>
      </Container>,
    ).toJSON();
    expect(screen).toMatchSnapshot();
  });
  it('renders correctly', () => {
    const screen = render(
      <Container>
        <Text>foo</Text>
      </Container>,
    );
    const content = screen.getByText('foo');
    expect(content.type).toBe(Text);
    expect(content.parent?.type).toBe('RCTSafeAreaView');
    expect(content.parent).toHaveStyle({flex: 1});
  });
  it('fills the available space', () => {
    const screen = render(
      <Container>
        <Text>foo</Text>
      </Container>,
    );
    expect(screen.getByText('foo').parent).toHaveStyle({flex: 1});
  });
  it('renders its children', () => {
    const screen = render(
      <Container>
        <Text>foo</Text>
        <Text>bar</Text>
      </Container>,
    );
    const contents = screen.getAllByText(/foo|bar/);
    expect(contents).toHaveLength(2);
    expect(contents[0].type).toBe(Text);
    expect(contents[0]).toHaveTextContent('foo');
    expect(contents[1].type).toBe(Text);
    expect(contents[1]).toHaveTextContent('bar');
  });
});
