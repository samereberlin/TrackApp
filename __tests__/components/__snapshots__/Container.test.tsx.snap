// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header matches the snapshot 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#f5f7f7",
      "flex": 1,
    }
  }
>
  <Text>
    foo
  </Text>
</RCTSafeAreaView>
`;
